README.markdown

```diff
@@ -449,6 +449,22 @@ If you want to stop processing jobs, but want to leave the worker running
 (for example, to temporarily alleviate load), use `USR2` to stop processing,
 then `CONT` to start it again.
 
+#### Signals on Heroku
+
+When shutting down processes, Heroku sends every process a TERM signal at the
+same time. By default this causes an immediate shutdown of any running job
+leading to frequent `Resque::TermException` errors.  For short running jobs, a simple
+solution is to give a small amount of time for the job to finish
+before killing it.
+
+To accomplish this set the following environment variables:
+
+* `RESQUE_PRE_SHUTDOWN_TIMEOUT` - The time between the parent receiving a shutdown signal (TERM by default) and it sending that signal on to the child process. Designed to give the child process
+time to complete before being forced to die.
+
+* `RESQUE_SHUTDOWN_SIGNAL` - The signal sent to the child process
```

Comment:
Make sure in testing you send the TERM signal to both the parent worker and the child process that actually executes the job.  That was the crux of the issue.  Just sending it to the parent process already has working behavior without this change but doesn't match what actually happens on heroku.